broker/src/main/java/org/apache/rocketmq/broker/processor/AdminBrokerProcessor.java

```diff
@@ -542,16 +543,28 @@ private synchronized RemotingCommand deleteTopic(ChannelHandlerContext ctx,
             }
         }
 
-        this.brokerController.getTopicConfigManager().deleteTopicConfig(requestHeader.getTopic());
-        this.brokerController.getTopicQueueMappingManager().delete(requestHeader.getTopic());
-        this.brokerController.getConsumerOffsetManager().cleanOffsetByTopic(requestHeader.getTopic());
-        this.brokerController.getPopInflightMessageCounter().clearInFlightMessageNumByTopicName(requestHeader.getTopic());
-        this.brokerController.getMessageStore().deleteTopics(Sets.newHashSet(requestHeader.getTopic()));
+        final Set<String> groups = this.brokerController.getConsumerOffsetManager().whichGroupByTopic(topic);
+        // delete pop retry topics first
```

Comment:
Because we already have the groups, I feel that it is better to check directly whether the pop retry topic exists in the TopicConfigManager, like
```java
for (String group : groups) {
    String popRetryTopic = KeyBuilder.buildPopRetryTopic(topic, group);
    if (brokerController.getTopicConfigManager().selectTopicConfig(popRetryTopic) != null) {
        deleteTopicInBroker(popRetryTopic);
    }
}
```

Reply:
ok, I’ve updated the code